packages/next/src/build/index.ts

```diff
@@ -1461,9 +1442,20 @@ export default async function build(
         }
       }
 
-      // For app directory, we run type checking after build.
-      if (appDir && !isCompileMode && !isGenerateMode) {
-        await startTypeChecking(typeCheckingOptions)
+      // Run type checking after the build as the build writes additional types.
+      if (!isCompileMode && !isGenerateMode) {
+        await startTypeChecking({
```

Comment:
Hmm do we want to move type checking for pages only as it may be a breaking change in users workflows as we won't be failing as early as we were previously.

Reply:
Okay let's keep it as-is then 👍